### ISSUE TITLE
Sharing in group folders doesn't work as expected

### ISSUE THREAD

**Issue by thomasrotter** [nextcloud/groupfolders#55]
* If I try to share a groupfolder with a user it's not accessible by the user I shared it with. In the activity feed the sharing activity can be seen - no file or folder icon just a grey square.

* If I share a groupfolder or a subfolder with a circle (app) it can be seen but the files and subfolders can't be accessed.

Tried it with Nextcloud 12.0.1 & 12.0.2

**Comment by geiseri** [comment]
I think I have the same issue.  I have a folder that has "share" privileges but that user cannot share any files inside that folder.  It says they are not allowed to "reshare".  Curiously the administrator user can share files in there just fine.

**Comment by putt1ck** [comment]
@thomasrotter can you confirm this is a group folder that you are then trying to further "normal" share with an individual?

**Comment by thomasrotter** [comment]
Exactly. I try to share a group folder or a subfolder of a group folder with an individual using the standard share functionality. Tried it again last Friday and still doesn't work - using 12.0.3 now.

**Comment by putt1ck** [comment]
Tested and confirmed as follows:

create group folder and add group B with full permissions;
as member of B, add subfolder X and share with user A not in the group;
user A *can* access share of X and view contents but cannot upload without errors despite having full permissions;
user A can access sub folder of X but cannot see contents added by other users;
user A can create (with errors) new items, including text files, uploaded file and folders (all with errors) but the items are not visible to other users only to user A;
all items (regardless of who created them) are visible in the server store in the expected folder.

Comments:
user A can comment on X but not items there in (specifically the attempt to add a comment never seems to complete, but the comment does actually appear if page is reloaded);
the comment issue affects members of B in the same way.

further:
user A can "share by link" folder X without error, but not a sub folder of X;
anyone accessing X via "share by link" gets expected permissions, including add, change and delete.

**Comment by putt1ck** [comment]
Thought following some poking around in the group folder code, and thinking about the purpose of group folders:

maybe the sharing functionality should either be limited to read only, or;
limited to share by link (which in basic testing seems to work).

**Comment by mc9625** [comment]
Any news on this bug? I still experience such problem. Actually in my experience there weren't all that errors, the only issue is that when a user (the one with you've shared the folder, not in the group owner of the folder, to be clear) uploads a file, as soon as he leaves the folder and reopen it, the file is disappeared, so there's a refresh issue I bet.
It seems to me rather a big issue.

**Comment by andrem1987** [comment]
We are running in the same Problem. We have a group-golder for internal users with full access rights. These users should be able to share subfolders and files with exernal users with full acces as well. Currently the external users only have read permissions. When trying to change or upload a file as an external user a "permission denied" message appears.

**Comment by kirosbr** [comment]
Up
Any news ?
I have still this bug with  nextcloud 14

**Comment by joseluis** [comment]
I'm having the issue with nextcloud 15. There's a general groupfolder that contains subfolders. Each of those subfolders is shared with an individual user (without no to the parent groupfolder), with permission to edit, and delete files given by the particular admin account that shared that folder with the user.

The problem is the users can't delete files from their own shared folder. They can create files, but no delete them after that point, and they get redownloaded.

**Comment by McAviti** [comment]
My experience with sharing group folders (now on 15.0.4): It works in principial, but I can confirm the not-able-to-delete issue. Additionally we had an increasing number of sync problems (with the current windows sync app) - so I cancelled all sharing and created specialized groups.

**Comment by Raqbit** [comment]
I can also say that sharing a subfolder of a group folder via a URL will show "unknown-file" icons for folders, like this: 
![image](https://user-images.githubusercontent.com/11144180/58384117-5c543300-7fde-11e9-9378-d0137d081615.png)

**Comment by crti** [comment]
On NC 16.0.1 I have the following: 
- as USER1 sharing a folder completely to a user group with USER2 by "Can edit" (Can reshare, Can create, Can change, Can delete) 
- login with USER2 I can create subfolders and upload files and have right to delete files/folders or share a folder/subfolder/file. 
- USER1 Gallery the same folder with subfolder seems also shared with all rights to the group
- USER2 however seems to not be able to share a single subfolder as Gallery to the public, upper folders work. Although gallery.cnf is there and listed for USER1 all other users and the public do not get the "i" button.

**Comment by levonti** [comment]
I have still this bug with nextcloud 16.0.4 and group folders 4.1.0

**Comment by vasyugan** [comment]
Still seeing this issue with NC 19.* and groupfolders 5.*

**Comment by fschrempf** [comment]
I just tested this on NC 21.0.0 and groupfolder 9.0.0 and I'm not able to reproduce these problems anymore.

I created a groupfolder and shared it with a group that user B is a member of and user A is not. As user B is created a subdirectory X and shared it with A (granting all permissions).

- User A is the able to see X and access it and sees all the subdirectories and files.
- User A can create/upload/edit/delete files and directories in X without errors and the changes are visible to user B.
- User A can write comments on files/directories in X and they are visible to B.
- User A can reshare X or a subdirectory of X with user C (who is also not in the group of the groupfolder)

May I assume this issue is solved and I can close it, or can anyone prove me wrong?
